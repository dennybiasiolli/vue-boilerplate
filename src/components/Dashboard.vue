<template>
  <div>
    <h1>Dashboard</h1>
    {{ message }}
  </div>
</template>

<script>
export default {
  data() {
    return {
      message: 'Hello Authenticated!'
    }
  }
}
</script>
