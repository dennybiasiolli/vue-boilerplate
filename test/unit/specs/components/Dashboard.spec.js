import { shallowMount } from '@vue/test-utils'
import Dashboard from '@/components/Dashboard'

describe('Dashboard', () => {
  let component, element

  beforeEach(() => {
    component = shallowMount(Dashboard)
    element = component.element
  })

  test('should match snapshot', () => {
    expect(element).toMatchSnapshot()
  })

  // Evaluate the results of functions in
  // the raw component options
  test('sets the correct default data', () => {
    expect(typeof Dashboard.data).toBe('function')
    const defaultData = Dashboard.data()
    expect(defaultData.message).toBe('Hello Authenticated!')
  })
})
