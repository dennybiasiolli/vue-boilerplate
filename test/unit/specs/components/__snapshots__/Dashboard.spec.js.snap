// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should match snapshot 1`] = `
<div>
  <h1>
    Dashboard
  </h1>
  
  Hello Authenticated!

</div>
`;
